### DISCUSSION TITLE
Replacing `onViewportBoxUpdate` functionality in v10

### DISCUSSION THREAD

**Discussion by hexcowboy** [motiondivision/motion#2162] [Q&A]
I have a component that works in framer motion up to v4. In v5, the `onViewportBoxUpdate` prop was removed.

```ts
function Item({ i, updatePosition, updateOrder, item }: Item) {
  const [isDragging, setDragging] = useState(false);
  const ref = useMeasurePosition((pos) => updatePosition(i, pos));

  return (
    <motion.div
      ref={ref}
      layout
      initial={false}
      drag
      onDragStart={() => setDragging(true)}
      onDragEnd={() => setDragging(false)}
      onViewportBoxUpdate={(_: AxisBox2D, delta: BoxDelta) => {
        if (isDragging) {
          updateOrder(i, delta.x.translate, delta.y.translate);
        }
      }}
    >
      <p>{item.id}</p>
    </motion.div>
  );
}
```

I believe its possible to refactor this without using `onViewportBoxUpdate`, but it depends on getting the `BoxDelta`'s `x.translate` and `y.translate`. I'm not entirely sure how to get the `BoxDelta` in v10.

Does anyone have any guidance or direction on how to migrate this component to v10?

[Codesandbox](https://codesandbox.io/p/github/hexcowboy/iphone-grid/main?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522panelType%2522%253A%2522TABS%2522%252C%2522id%2522%253A%2522cli5393hg000b2e6h954xm8wo%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522panelType%2522%253A%2522TABS%2522%252C%2522id%2522%253A%2522cli5393hg000d2e6hrumhlxt8%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cli5393hg000b2e6h954xm8wo%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cli5393hg000a2e6hv6y649f6%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%257D%255D%252C%2522id%2522%253A%2522cli5393hg000b2e6h954xm8wo%2522%252C%2522activeTabId%2522%253A%2522cli5393hg000a2e6hv6y649f6%2522%257D%252C%2522cli5393hg000d2e6hrumhlxt8%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cli5393hg000c2e6hybb1ubz5%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522PROJECT_SETUP%2522%257D%255D%252C%2522id%2522%253A%2522cli5393hg000d2e6hrumhlxt8%2522%252C%2522activeTabId%2522%253A%2522cli5393hg000c2e6hybb1ubz5%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by kschmott** [comment]
Did you figure this out? Facing the same issue.

**Reply by hexcowboy** [reply]
I have not. Still using v4
